src/util/records.py

```diff
@@ -1147,6 +1148,8 @@ def add_ref(ref):
             doc = lxml.etree.parse(fp)
             for node in doc.xpath(xpath):
                 found = True
+                if xml_filename is None:
```

Comment:
I wonder if we now need the `found` variable? We can keep the xml_filename as the "found" signal whenever it's set. Maybe named `found_in_xml_file`?

There is a slight subtlety. We do not break the search loop when found. Thus in practice we keep searching and adding nodes to our new root. This means that we will be lying if we keep the same filename when loading our new root. Maybe we should have a `new_root` per `found_in_xml_file`, a sort of map `filename->new_root`. We should double check the standard behavior if we really care about this.